src/discussions/posts/post/Post.jsx

```diff
@@ -86,31 +86,54 @@ function PostHeader({
           </span>
         </div>
       </div>
-      <div className="d-flex mr-3">
-        <Icon src={QuestionAnswer} />
-        <span style={{ minWidth: '2rem' }}>
-          {post.commentCount}
-        </span>
-      </div>
+      {preview
+        ? (
+          <div className="d-flex">
+            <Icon src={QuestionAnswer} />
+            <span style={{ minWidth: '2rem' }}>
+              {post.commentCount}
+            </span>
+          </div>
+        )
+        : <ActionsDropdown commentOrPost={post} actionHandlers={actionHandlers} />}
     </div>
   );
 }
 
 PostHeader.propTypes = {
   intl: intlShape.isRequired,
   post: postShape.isRequired,
+  preview: PropTypes.bool.isRequired,
+  actionHandlers: PropTypes.objectOf(PropTypes.func).isRequired,
 };
 
 function Post({
   post,
+  preview,
   intl,
 }) {
+  const location = useLocation();
+  const history = useHistory();
   const dispatch = useDispatch();
+  const actionHandlers = {
+    [EditActions.EDIT_CONTENT]: () => history.push(`${location.pathname}/edit`),
+    [EditActions.DELETE]: () => dispatch(removeThread(post.id)),
```

Comment:
On Delete is deleting post but the UI stays in the loading state and the post id remains appended in url. I think we should redirect back user to the main page after deleting the post. what do you say?

Reply:
The delete code will be changed once we have a confirmation step, right now this is a placeholder till we have a mockup for that.

Reply:
can you please add a TODO item in the code so we can proceed with closing this thread?